src/librbd/mirror/snapshot/CreatePrimaryRequest.cc

```diff
@@ -205,6 +205,10 @@ void CreatePrimaryRequest<I>::unlink_peer() {
         unlink_snap_id = 0;
         continue;
       }
+      if (info->mirror_peer_uuids.count(peer) == 0
+          && info->mirror_peer_uuids.size() != 0) {
+        continue;
+      }
```

Comment:
Sorry, I think I misled you.

Yes, this is what I meant. But looking at it now I see a problem. We want to keep `max_snapshot` snapshots belonging to the peer. But here we also increase the `count` if the snapshot is "orphaned".

It looks like we have to have a special condition, i.e. check if `info->mirror_peer_uuids` is empty and then set `peer_uuid = peer`, `snap_id = snap_it.first` and break. So we will always prefer to unlink (remove) "orphaned" snapshots first. Or feel free to try in some other way if you like.

Reply:
No worries, I think my previous attempts would have work this way but an early condition like this seems cleaner!